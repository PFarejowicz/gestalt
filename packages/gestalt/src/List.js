// @flow

import * as React from 'react';
import Box from './Box.js';
import ListItem from './ListItem.js';

type Props = {|
  items: Array<{|
    onClick: ({
      event:
        | SyntheticMouseEvent<HTMLDivElement>
        | SyntheticKeyboardEvent<HTMLDivElement>,
    }) => void,
    children: React.Node,
  |}>,
|};

export default function List({ items }: Props) {
  return (
    <Box column={12} paddingY={4} role="list">
      {items.map((item, index) => (
        <ListItem key={index} onClick={item.onClick}>
          {item.children}
        </ListItem>
      ))}
    </Box>
  );
}
