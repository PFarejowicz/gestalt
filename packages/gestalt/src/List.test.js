// @flow

import React from 'react';
import renderer from 'react-test-renderer';
import List from './List.js';

test('List renders', () => {
  const component = renderer.create(
    <List items={[{ onClick: () => {}, children: 'Item' }]} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
