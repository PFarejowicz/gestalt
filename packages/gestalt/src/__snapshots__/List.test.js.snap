// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders 1`] = `
<div
  className="box paddingY4 xsCol12"
  role="list"
>
  <div
    className="box paddingX1 paddingY1 whiteBg"
    role="listitem"
  >
    <div
      className="touchable pointer square fullWidth"
      onClick={[Function]}
      onKeyPress={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="box paddingX3 paddingY1 xsCol12"
      >
        <div
          className="Text fontSize3 darkGray alignLeft breakWord fontStyleNormal fontWeightBold"
        >
          Item
        </div>
      </div>
    </div>
  </div>
</div>
`;
